src/compute-shortname.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * Module that exports a function that takes a URL as input and computes a
+ * meaninful shortname, family name and level for it, when appropriate.
```

Comment:
s/meaninful/meaningful

Looking at [Version Management in W3C Technical Reports](https://www.w3.org/2005/05/tr-versions), the (arguably somewhat ambiguous) taxonomy seems to be that shortname is the non-leveled version; so maybe we should talk about "leveled shortname" and "shortname" (or "bare shortname", or...)

Reply:
Ah, I hadn't looked at that, good if we have some sort of official definition for `shortname`. I'll update the code accordingly. That would actually match the terminology used in Shepherd. And I can probably also follow Shepherd's naming convention for the "leveled shortname" and simply call the propery `name` (alternatives could be `leveledShortname` or perhaps `shortnameN` but neither looks appealing)